app/decryptshards.js

```diff
@@ -0,0 +1,228 @@
+                return indices.indexOf(index) !== -1;
+            });
+
+            const firstKey = keysToRecover[0];
+
+            console.log('\nYou must now enter your encryption passwords. Note that the order does not matter for these passwords. You only must enter ' + firstKey.m + ' of the ' + firstKey.n + ' passwords you created.\n');
+
+
+            // Get the passwords
+            return getPassword(0, firstKey.m, firstKey.seedShares);
+        })
+        .then(function() {
+            // For each key we want to recover, decrypt the shares, recombine
+            // into a seed, and produce the xprv, validating against existing xpub.
+            const recoveredKeys = keysToRecover.map(function(key) {
+                const shares = passwords.map(function(p, i) {
+                    console.log('Decrypting Key Part #' + i);
+                    return sjcl.decrypt(p.password, key.seedShares[p.shareIndex]);
+                });
+                let seed;
+                if (shares.length === 1) {
+                    seed = shares[0];
+                } else {
+                    seed = secrets.combine(shares);
+                }
+
+                // here handle xlm differently
+
+                if(args.type === 'xlm') {
+
+                    return {seed: seed};
+
+                } else {
+                    const extendedKey = bitcoin.HDNode.fromSeedHex(seed);
+                    const xpub = extendedKey.neutered().toBase58();
+                    const xprv = args.verifyonly ? undefined : extendedKey.toBase58();
+                    if (!args.verifyonly && xpub !== key.xpub) {
+                        throw new Error("xpubs don't match for key " + key.index);
+                    }
+                    return {
+                        index: key.index,
+                        xpub: xpub,
+                        xprv: xprv,
+                        seed: seed
+                    };
+                }
+
+
+            });
+            if(onFinish) {
+                onFinish(recoveredKeys);
+            }
+        });
+};
+
+/**
+ * Prompt the user to upload all the key shares from separate files
+ * @param all : when true, the prompt will continue until all N shares are uploaded.
+ * When false, the prompt will stop once M shares have been uploaded.
+ *
+ * returns a list containing 1 json object
+ */
+const uploadKeyShares = function(all, type) {
+    console.log("\nIt is now time to upload the key shares you wish to use.\n");
+    const currentPrvs= [];
+    let m = null;
+    let n = null;
+    let xpub = null;
+    while(true) {
+        const inputfile = prompt("Enter the file path containing a key share: ");
+        try {
+            const file = fs.readFileSync(inputfile);
+            const json = JSON.parse(file);
+
+            // Enforce file contents are correct
+            if(!json.m) throw new Error('File does not contain the required field: m');
```

Comment:
Can we clean this up a little? I think you could make this a loop -- something like

```
['m', 'n', /* etc */].forEach( requiredField => {
    if (!json[requiredField]) {
        throw new Error('file does not contain ' + requiredField)
    }
})
```